### ISSUE TITLE
Block controls: Show tooltip on hover, keyboard focus

### ISSUE THREAD

**Issue by aduth** [WordPress/gutenberg#531]
Block controls have a `title` property associated with them ([example](https://github.com/WordPress/gutenberg/blob/36bbf962fac4edf709c091f62e0b593cebaac378/blocks/library/text/index.js#L27)). Currently this is only assigned as the [`aria-label` of the rendered `IconButton` component](https://github.com/WordPress/gutenberg/blob/36bbf962fac4edf709c091f62e0b593cebaac378/editor/components/icon-button/index.js#L17). This isn't exposed when hovering the button, so it's hard to know what action is associated with a particular block control.

- Should we show the title as a tooltip?
- How does it look?
- On what actions should it appear? Hover, focus?

**Comment by iseulde** [comment]
If it's essential information, it should probably not be shown on hover, and if it isn't, should it be shown at all? (I know this is different in the current editor)

**Comment by jasmussen** [comment]
For some of the buttons, like the Action Buttons, I think it'd be nice with an immediate tooltip, like this:

![screen shot 2017-04-28 at 18 32 11](https://cloud.githubusercontent.com/assets/1204802/25537969/0ec8544c-2c41-11e7-8c7d-4030837ee209.png)

For the majority of buttons, like Bold/Italic, etc. —  it's probably fine with a title attribute and the stock browser tooltip.

**Comment by afercia** [comment]
> On what actions should it appear? Hover, focus?

I'd say on both, with the caveat that when a control gets clicked or activated with the keyboard, is stays "focused", unless focus is moved somewhere else. So there should be a way to hide the tooltip after the control activation.

Worth  noting visual tooltips can be built in an accessible way and using just CSS, using the content property and targeting the `aria-label` attribute. See for example what GitHub does in this same page (see for example Notifications, Create new, and profile on the top right of this page).

**Comment by aduth** [comment]
Thanks for pointing to the GitHub example @afercia , I think we could make something like this work for our case. Based on the styles, it appears they're capturing the `:hover`, `:active`, and `:focus` pseudo classes. I'll plan to experiment with these in the editor.

**Comment by afercia** [comment]
Worth noting that tooltips were also mentioned in the initial accessibility recommendations, see point 8 in #297 

Furthermore, they were often mentioned in various "Testing feedback" reported by @karmatosed, see #1546, #1547, #1549, and #1550.

**Comment by afercia** [comment]
Description of the ARIA tooltip pattern, still in development:
https://www.w3.org/TR/wai-aria-practices-1.1/#tooltip

**Comment by afercia** [comment]
Worth noting that also the block toolbars don't have any tooltip at the moment. Not even the "Classic Text" one, that uses the native TinyMCE controls:

![screen shot 2017-07-25 at 20 22 06](https://user-images.githubusercontent.com/1682452/28630778-7f3cd006-722b-11e7-8c26-c7f4fde845dc.png)

While the same controls in the old editor do have tooltips:

![screen shot 2017-07-25 at 20 21 43](https://user-images.githubusercontent.com/1682452/28630799-8ba1faa6-722b-11e7-8233-9361a394f2c3.png)

Right now, they're also used to inform users about the associated keyboard shortcuts. That seems something valuable to me.

**Comment by paaljoachim** [comment]
It would also be good to add tooltips to the image toolbar so the user can understand what each icon does before clicking it.

**Comment by aduth** [comment]
See also in-progress work at #839, which was made difficult by positioning, impact of style cascade. With refactoring to `Popover` component in #2160, I suspect it will become very easy to implement tooltips.

**Comment by aduth** [comment]
Considering this closed by #2293
